api/v3/Contribution.php

```diff
@@ -533,6 +533,7 @@ function civicrm_api3_contribution_completetransaction(&$params) {
   if (isset($params['payment_processor_id'])) {
     $input['payment_processor_id'] = $params['payment_processor_id'];
   }
+  $input['total_amount'] = CRM_Utils_Array::value('total_amount', $params);
```

Comment:
What would be the use case to update the total amount for a  contribution?

Reply:
@pradpnayak So we're not updating the total_amount but we need to pass it as a parameter so the later functions can check if the amount is 0.